packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -606,6 +637,18 @@ export class AppSyncModelVisitor<
     return intermediateModel;
   }
 
+  protected generateIntermediateModelSortKeyFieldName(model: CodeGenModel, sortKeyField: CodeGenField): string {
+    const modelName = model.name.charAt(0).toLocaleLowerCase() + model.name.slice(1);
+    return `${modelName}${sortKeyField.name}`;
+  }
+
+  protected getSortKeyFields(model: CodeGenModel): CodeGenField[] {
+    const keyDirective = model.directives.find(d => d.name === 'key' && !d.arguments.name);
+    return keyDirective
+      ? (keyDirective.arguments.fields as string[]).slice(1).map(fieldName => model.fields.find(f => f.name === fieldName)!)
```

Comment:
Why are we ignoring the first argument via `slice`? Is that key name argument?

Reply:
In old `@key` without name the first field is defined as primary key